### PR TITLE
Fix time issues for doc build on circleci

### LINKED ISSUE #1420: Docs take a long time to build
As of late our docs take ~14 minutes to build on circle-ci whereas they took about 6 minutes to build in the previous release. The root cause of this slow-down seems to be that woodwork is inferring some categorical variables as text which then causes AutoML to use the TextFeaturizer. However, even if ww fixes the categorical vs text inference, the time to build the docs will inevitably increase as we write more documentation. This makes it hard for developers to iterate on the docs locally.

Possible solutions:

- Add some hidden code in the notebook that would skip long running computations.
- Have nb-sphinx or read the docs cache long-running computations.

### PR DESCRIPTION
**Pull request by bchen1116**
partially addresses #1420 

Doc build on circleci now takes ~6 minutes, and with the additional tasks that circleci runs, it takes a total of ~8-9 minutes to complete. 

This is separate from building docs that we use for our documentation. Will put up a separate PR for fixing ReadTheDocs when we get feedback or find a solution to speed up the RTD build, in which case, will then close #1420.

### CODE REVIEW

File: docs/source/demos/fraud.ipynb [hunk 1]
```diff
@@ -174,16 +174,7 @@
...
-    "Finally, we retrain the best pipeline on all of the training data and evaluate on the holdout data."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "best_pipeline.fit(X_train, y_train)"
```
Comment (bchen1116) [inline docs/source/demos/fraud.ipynb:186]: No longer need to fit best pipelines

File: docs/source/user_guide/model_understanding.ipynb [hunk 2]
```diff
@@ -225,9 +225,7 @@
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "scrolled": false
-   },
+   "metadata": {},
```
Comment (freddyaboulton) [inline docs/source/user_guide/model_understanding.ipynb:228]: No need to respond lol but I wonder why this diff happened?

File: docs/source/user_guide/automl.ipynb [hunk 3]
```diff
@@ -363,7 +363,7 @@
     "from evalml.pipelines.utils import make_pipeline\n",
     "import woodwork as ww\n",
     "\n",
-    "X, y = load_fraud()\n",
+    "X, y = load_fraud(n_rows=1000)\n",
```
Comment (freddyaboulton) [inline docs/source/user_guide/automl.ipynb:366]: 🥳

File: docs/Makefile [hunk 4]
```diff
@@ -18,7 +18,7 @@ clean:
 
 .PHONY: html
 html:
-	$(SPHINXBUILD) -b html $(ALLSPHINXOPTS) $(BUILDDIR)/html $(SPHINXOPTS)
+	$(SPHINXBUILD) -b html $(ALLSPHINXOPTS) $(BUILDDIR)/html $(SPHINXOPTS)  -j 'auto'
```
Comment (dsherry) [inline docs/Makefile:21]: Its cool that this sped up the `build_docs` job :) out of curiosity do you know by how much?

File: docs/source/demos/fraud.ipynb [hunk 5]
```diff
@@ -174,16 +174,7 @@
...
+    "Finally, since the best pipeline is already trained, we evaluate it on the holdout data."
```
Comment (dsherry) [inline docs/source/demos/fraud.ipynb:177]: Thanks for catching these!

File: docs/source/user_guide/automl.ipynb (hunk 3 above)
Comment (dsherry) [inline docs/source/user_guide/automl.ipynb:366]: 👍  I think this brings the runtime here down from ~30sec to almost nothing

File: docs/source/user_guide/model_understanding.ipynb [hunk 6]
```diff
@@ -254,9 +252,7 @@
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "scrolled": false
```
Comment (dsherry) [inline docs/source/user_guide/model_understanding.ipynb:258]: Is this important? Does it affect the visual output?

**Review by dsherry — APPROVED** [review]
🚢  👏

**Review by freddyaboulton — APPROVED** [review]
@bchen1116 This is awesome! Looks like the old build time was like 15 minutes so this is a big improvement!

File: docs/Makefile (hunk 4 above)
Comment (bchen1116) [reply docs/Makefile:21]: It goes from about 8-9 minutes before this change to ~6 minutes after

File: docs/source/demos/fraud.ipynb (hunk 1 above)
Comment (angela97lin) [reply docs/source/demos/fraud.ipynb:186]: Nice!!

**Review by angela97lin — APPROVED** [review]
Woot, LGTM! 🥳

File: docs/source/user_guide/model_understanding.ipynb (hunk 6 above)
Comment (bchen1116) [reply docs/source/user_guide/model_understanding.ipynb:258]: This is the output before the change
image

And this is the output after
image

They seem the same to me. @freddyaboulton fyi

**Pull request merged**